12-offer-encoding.md

```diff
@@ -0,0 +1,870 @@
+    2. data:
+        * [`...*byte`:`data`]
+    1. type: 6 (`offer_currency`)
+    2. data:
+        * [`...*utf8`:`iso4217`]
+    1. type: 8 (`offer_amount`)
+    2. data:
+        * [`tu64`:`amount`]
+    1. type: 10 (`offer_description`)
+    2. data:
+        * [`...*utf8`:`description`]
+    1. type: 12 (`offer_features`)
+    2. data:
+        * [`...*byte`:`features`]
+    1. type: 14 (`offer_absolute_expiry`)
+    2. data:
+        * [`tu64`:`seconds_from_epoch`]
+    1. type: 16 (`offer_paths`)
+    2. data:
+        * [`...*blinded_path`:`paths`]
+    1. type: 18 (`offer_issuer`)
+    2. data:
+        * [`...*utf8`:`issuer`]
+    1. type: 20 (`offer_quantity_max`)
+    2. data:
+        * [`tu64`:`max`]
+    1. type: 22 (`offer_issuer_id`)
+    2. data:
+        * [`point`:`id`]
+    1. type: 80 (`invreq_chain`)
+    2. data:
+        * [`chain_hash`:`chain`]
+    1. type: 82 (`invreq_amount`)
+    2. data:
+        * [`tu64`:`msat`]
+    1. type: 84 (`invreq_features`)
+    2. data:
+        * [`...*byte`:`features`]
+    1. type: 86 (`invreq_quantity`)
+    2. data:
+        * [`tu64`:`quantity`]
+    1. type: 88 (`invreq_payer_id`)
+    2. data:
+        * [`point`:`key`]
+    1. type: 89 (`invreq_payer_note`)
+    2. data:
+        * [`...*utf8`:`note`]
+    1. type: 90 (`invreq_paths`)
+    2. data:
+        * [`...*blinded_path`:`paths`]
+    1. type: 240 (`signature`)
+    2. data:
+        * [`bip340sig`:`sig`]
+
+## Requirements for Invoice Requests
+
+The writer:
+  - if it is responding to an offer:
+    - MUST copy all fields from the offer (including unknown fields).
+    - if `offer_chains` is set:
+      - MUST set `invreq_chain` to one of `offer_chains` unless that chain is bitcoin, in which case it MAY omit `invreq_chain`.
+    - otherwise:
+      - if it sets `invreq_chain` it MUST set it to bitcoin.
+    - MUST set `signature`.`sig` as detailed in [Signature Calculation](#signature-calculation) using the `invreq_payer_id`.
+    - if `offer_amount` is not present:
+      - MUST specify `invreq_amount`.
+    - otherwise:
+      - MAY omit `invreq_amount`.
+      - if it sets `invreq_amount`:
+        - MUST specify `invreq_amount`.`msat` as greater or equal to amount expected by `offer_amount` (and, if present, `offer_currency` and `invreq_quantity`).
+    - MUST set `invreq_payer_id` to a transient public key.
+    - MUST remember the secret key corresponding to `invreq_payer_id`.
+    - if `offer_quantity_max` is present:
+      - MUST set `invreq_quantity` to greater than zero.
+      - if `offer_quantity_max` is non-zero:
+        - MUST set `invreq_quantity` less than or equal to `offer_quantity_max`.
+    - otherwise:
+      - MUST NOT set `invreq_quantity`
+  - otherwise (not responding to an offer):
+    - MUST set `offer_description` to a complete description of the purpose of the payment.
+    - MUST set (or not set) `offer_absolute_expiry` and `offer_issuer` as it would for an offer.
```

Comment:
Yes, though only for invoice request's without an offer (e.g., a scannable refund). Otherwise, it must be reflected back as any other TLV in an offer.